tests/test_server.py

```diff
@@ -560,13 +557,8 @@ class TestDevice(Device):
         def get_prop(self):
             return self.prop
 
-    with DeviceTestContext(TestDevice,
-                           properties={'prop': value},
-                           process=True) as proxy:
-        assert_close(proxy.get_prop(), expected(value))
-
```

Comment:
This test case was important.  It checks that mandatory properties that have values specified in the database, work correctly.

Here's is a version that tests both cases:
- works if a value is specified in the database
- raises an error if not specified in the database

We now have to split it into two different tests.

```python
def test_mandatory_device_property_with_db_value_succeeds(
        typed_values, server_green_mode):
    dtype, values, expected = typed_values
    patched_dtype = dtype if dtype != (bool,) else (int,)
    default, value = values[:2]

    class TestDevice(Device):
        green_mode = server_green_mode

        prop = device_property(dtype=dtype, mandatory=True)

        @command(dtype_out=patched_dtype)
        def get_prop(self):
            return self.prop

    with DeviceTestContext(TestDevice,
                           properties={'prop': value}) as proxy:
        assert_close(proxy.get_prop(), expected(value))


def test_mandatory_device_property_without_db_value_fails(
        typed_values, server_green_mode):
    dtype, _, _ = typed_values

    class TestDevice(Device):
        green_mode = server_green_mode
        prop = device_property(dtype=dtype, mandatory=True)

    with pytest.raises(DevFailed) as context:
        with DeviceTestContext(TestDevice):
            pass
    assert 'Device property prop is mandatory' in str(context.value)

```